elements/lisk-cryptography/src/ed.ts

```diff
@@ -111,26 +99,21 @@ export const signMessageWithPrivateKey = (
 	};
 };
 
-export const verifyMessageWithPublicKey = ({
-	message,
-	publicKey,
-	signature,
-}: SignedMessageWithPrivateKey): boolean => {
-	const msgBytes = digestMessage(message);
-
+export const verifyMessageWithPublicKey = (
+	{ message, publicKey, signature }: SignedMessageWithPrivateKey,
+	tag = MESSAGE_TAG_NON_PROTOCOL_MESSAGE,
```

Comment:
why not adding this to the object?

Reply:
also if we support buffer above, we should support it here as well

Reply:
> why not adding this to the object?

Because the object type is called `SignedMessageWithPrivateKey` and IMO `tag` has nothing to do with it.

Also, if we modify the `SignedMessageWithPrivateKey` to include a `tag` property, then we would need to make a modification also in `signMessageWithPrivateKey()` to return that `tag`.

Speaking of, I would definitely prefer to simplify `signMessageWithPrivateKey()` to return only a `signature`. `message` is already provided as input argument, so I don't see a need to return it. Deriving and returning a `publicKey` is IMO not what should be concern of this function.

As for `verifyMessageWithPublicKey()` arguments, my proposal is to get rid of the `SignedMessageWithPrivateKey` object as the argument, and provide all 4 arguments as individual variables instead.

What do you think?

Reply:
tag is part of `SignedMessageWithPrivateKey` because tag is always signed with the message and without knowing that it's not possible to verify the signature?

I agree that your func signature idea is better, but to avoid breaking change at this point, it would be better to keep it backward compatible

Reply:
Maybe that's what Shu meant in his previous message: Isn't `tag` a redundant argument since it will always be equal to `MESSAGE_TAG_NON_PROTOCOL_MESSAGE`?

Reply:
It will not always have that value. It is a default value for the tag, if the value is not provided when invoking the function. User can provide a custom value for the tag, in order to verify any message.

My original implementation did not allow users to specify the tag. Then @shuse2 [requested to introduce that functionality](https://github.com/LiskHQ/lisk-sdk/pull/8805#discussion_r1284002053).